wfl/Controllers/RestaurantsController.cs

```diff
@@ -23,33 +23,43 @@ public RestaurantsController(IDBManager dbManager)
             {
                 return BadRequest();
             }
-            Restaurant r = await DBManager.InsertRestaurantAsync(restaurant);
+            Restaurant r = new Restaurant()
+            {
+                Name = restaurant.Name,
+                MenuLink = restaurant.MenuLink
+            };
+            r = await DBManager.InsertEntityAsync(r);
             return Created(r.Id.ToString(), r);
         }
 
         [Route("{id}")]
         [HttpPut]
         public async Task<IHttpActionResult> UpdateAsync([FromUri] string id, [FromBody] RestaurantDto restaurant)
         {
-            if (string.IsNullOrEmpty(id) || await DBManager.GetRestaurantWithIdAsync(id) == null || string.IsNullOrEmpty(restaurant?.Name) || (restaurant.MenuLink != null && !Uri.IsWellFormedUriString(restaurant.MenuLink.ToString(), UriKind.Absolute)))
+            if (string.IsNullOrEmpty(id) || await DBManager.GetEntityByIdAsync<Restaurant>(id) == null || string.IsNullOrEmpty(restaurant?.Name) || (restaurant.MenuLink != null && !Uri.IsWellFormedUriString(restaurant.MenuLink.ToString(), UriKind.Absolute)))
             {
                 return BadRequest();
             }
-            return Ok(await DBManager.UpdateRestaurantAsync(id, restaurant));
+            Restaurant r = new Restaurant()
```

Comment:
Why are we creating a new restaurant and then setting the values based on what was put?  Really we should be loading the restaurant from the database and then updating it's values, and then saving it back.  While in this particular case it won't make much of a difference, in the case where there is actual business logic being executed on the entity, it would be important to allow that logic to run on the existing item based on it's current state.